### PR TITLE
boost: add official patches

### PR DESCRIPTION
**Pull request by gjasny**
Add missing upstream patches for Boost 1.73 - 1.80.

Fixes: #2163

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
This is a very old problem 😄 Since these issues are already fixed in newer versions of boost we have not accept them.
It's also a problem as these are not tested by boost so we have no guarantee they work in harmony.

https://github.com/conan-io/conan-center-index/pull/13053
https://github.com/conan-io/conan-center-index/pull/6132
https://github.com/conan-io/conan-center-index/pull/3918
https://github.com/conan-io/conan-center-index/pull/436

Given one of the patches is so massive it too big, you will need to delete that for the build to at least pass.
The other 73 patch seems really odd, and it such an old version, I'd suggest dropping that too.

As for the two for 80, they are for portability and more inline without patching rules so I'd be include to accept those since they are very small.

**Comment by SSE4** [comment]
@grafikrobot can you comment on this one?

**Comment by gjasny** [comment]
Hello,

thanks for the quick feedback.

> Given one of the patches is so massive it too big, you will need to delete that for the build to at least pass. The other 73 patch seems really odd, and it such an old version, I'd suggest dropping that too.

I was working on that because I went over the open issues. If you don't intend to accept it, could you please close #2163? 

> As for the two for 80, they are for portability and more inline without patching rules so I'd be include to accept those since they are very small.

I removed the patches for 1.73 and only kept the ones for 1.80.

Thanks,
Gregor

**Comment by prince-chrismc** [comment]
> If you don't intend to accept it, could you please close

Done!

File: recipes/boost/all/conandata.yml [hunk 1]
```diff
@@ -88,6 +88,14 @@ patches:
       patch_description: "On POSIX systems that don't support *at APIs, compilation fails due to a missing include"
       patch_type: "official"
       patch_source: "https://github.com/boostorg/filesystem/issues/250"
+    - patch_file: "patches/1.80.0-0005-config-libcpp15.patch"
+      patch_description: "Support for libcpp15 which removes std::unary_function and std::binary_function."
```
Comment (prince-chrismc) [inline recipes/boost/all/conandata.yml:92]:
```suggestion
      patch_description: "Support for libcpp15 which removes std::unary_function and std::binary_function."
      patch_type: "portability"
      patch_source: "https://www.boost.org/patches/1_80_0/0005-config-libcpp15.patch"
```

Comment (gjasny) [reply recipes/boost/all/conandata.yml:92]: Applied.

**Review by jwillikers — APPROVED** [review]

**Pull request merged**